src/server/context.ts

```diff
@@ -276,11 +276,12 @@ export class ServerContext {
       // to their non-trailing slash counterpart.
       // Ex: /about/ -> /about
       const url = new URL(req.url);
-      if (url.pathname.length > 1 && url.pathname.endsWith("/")) {
+      const pathname = url.pathname;
+      if (pathname.length > 1 && pathname.endsWith("/")) {
```

Comment:
`const { pathname } = url;
  if (pathname && pathname.endsWith("/")) {
`
Should have the same effect.